### PR TITLE
Add GitHub CI and Validation workflow

### PR DESCRIPTION
**Pull request by josueetcom**
This PR adds two workflows to the project:
- A PR validation workflow that:
  - **audits** our dependencies to check for security vulnerabilities
  - ensures the application **builds**
  - ensures **tests** pass
  - **lints** the project for formatting issues according to the [Airbnb React/JSX Style Guide](https://airbnb.io/javascript/react/)
- A CI workflow that deploys the `main` branch to Firebase once a PR is merged.

We'll be following the [Airbnb React/JSX Style Guide](https://airbnb.io/javascript/react/). Check it out to see what it's about, or if you prefer video, check out [this video playlist](https://scrimba.com/p/playlist-25/cast-1471) explaining it.

The linter's been set up to detect most formatting issues. Once this PR is merged, you'll be able to run `yarn lint` to see what formatting issues there are and `yarn lint --fix` to fix most of them. I've fixed linting errors in our current source code.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+build/
```
Comment (josueetcom) [inline .eslintignore:1]: [ESLint](https://eslint.org/) is what's used find and fix problems in JavaScript code. This file tells ESLint to ignore the `build/`, which is generated when `yarn build` is run.

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+yarn.lock linguist-generated=true
```
Comment (josueetcom) [inline .gitattributes:1]: `yarn.lock` is generated. We're not interested in reviewing it, nor in having it count towards our lines of code in anyway.

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Workflow for deploying app to Firebase
```
Comment (josueetcom) [inline .github/workflows/deploy.yml:1]: For an overview on GitHub actions, see [this article](https://docs.github.com/en/actions/getting-started-with-github-actions/overview)

File: firebase.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (josueetcom) [inline firebase.json:1]: If you're curious about what this file does exactly, check [these docs](https://firebase.google.com/docs/hosting/full-config) out.

**Review by edwingl10 — APPROVED** [review]

**Pull request merged**